### PR TITLE
synchronous tidb decimal.go add NewDecFromString and NewDecFromFloat function

### PR DESCRIPTION
**Pull request by luky116**
**What this PR does**:
add NewDecFromString and NewDecFromFloat function

### CODE REVIEW

File: math/big/decimal.go [hunk 1]
```diff
@@ -2291,6 +2298,13 @@ func NewDecFromStringForTest(s string) *Decimal {
 	return dec
 }
 
+// NewDecFromString creates a Decimal from string.
+func NewDecFromString(s string) (*Decimal, error) {
+	dec := new(Decimal)
```
Comment (AlexStocks) [inline math/big/decimal.go:2303]: ec := &Decimal{}

File: math/big/decimal.go [hunk 2]
```diff
@@ -2283,6 +2283,13 @@ func NewDecFromFloatForTest(f float64) *Decimal {
 	return dec
 }
 
+// NewDecFromFloat creates a Decimal from float.
+func NewDecFromFloat(f float64) (*Decimal, error) {
+	dec := new(Decimal)
```
Comment (AlexStocks) [inline math/big/decimal.go:2288]: ec := &Decimal{}

不要使用 new

File: math/big/decimal.go (hunk 1 above)
Comment (luky116) [reply math/big/decimal.go:2303]: ok, i'll fix

File: math/big/decimal.go (hunk 2 above)
Comment (luky116) [reply math/big/decimal.go:2288]: ok, i'll fix

**Review by AlexStocks — APPROVED** [review]

**Review by wenxuwan — APPROVED** [review]

File: math/big/decimal.go [hunk 3]
```diff
@@ -2283,6 +2283,13 @@ func NewDecFromFloatForTest(f float64) *Decimal {
...
 
+// NewDecFromFloat creates a Decimal from float.
+func NewDecFromFloat(f float64) (*Decimal, error) {
+	dec := &Decimal{}
+	err := dec.FromFloat64(f)
+	return dec, err
+}
+
 // NewDecFromStringForTest creates a Decimal from string, as it returns no error, it should only be used in test.
 func NewDecFromStringForTest(s string) *Decimal {
```
Comment (wenxuwan) [inline math/big/decimal.go:2294]: now we can use  NewDecFromString for ut test, the method can remove now ?

**Review by baerwang — APPROVED** [review]

File: math/big/decimal.go (hunk 3 above)
Comment (luky116) [reply math/big/decimal.go:2294]: i'll remove it

**Pull request closed without merging**